api/rest/restapi.go

```diff
@@ -1025,7 +1016,39 @@ func (api *API) parseCidOrError(w http.ResponseWriter, r *http.Request) types.Pi
 		return types.PinSerial{Cid: ""}
 	}
 
-	return api.parsePinSerial(w, r, hash)
+	pin := types.PinSerial{
+		Cid:  hash,
+		Type: uint64(types.DataType),
+	}
+
+	queryValues := r.URL.Query()
+	name := queryValues.Get("name")
+	pin.Name = name
+	pin.MaxDepth = -1 // For now, all pins are recursive
+	rplStr := queryValues.Get("replication")
```

Comment:
With this PR, PinHandler, UnpinHandler and code for pin/unpin using cid is redundant code(since all those things can be done by pinpath/unpinpath). This was my attempt to keep that code separately from PinPath/UnpinPath. 
Are we going to get rid of that code(Pin/Unpin with cid)

Reply:
But we are not going to get rid of that code. I have already explained it:

POST pins/<cid> needs to keep working as before
POST pins/{ipfs,ipld,ipns}/... is the new feature which allows to provide a path and resolves them

we are not going to remove pins/<cid> as a separate route, but they of course can share things like params parsing.

And even if it wasn't so, that is no reason to duplicate code from a function.